rhino/src/main/java/org/mozilla/javascript/LambdaAccessorSlot.java

```diff
@@ -0,0 +1,129 @@
+package org.mozilla.javascript;
+
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+/**
+ * A specialized property accessor using lambda functions, similar to {@link LambdaSlot}, but allows
+ * defining properties with getter and setter lambdas that require access to the owner object
+ * ('this'). This enables the implementation of properties that can access instance fields of the
+ * owner.
+ *
+ * <p>Unlike {@link LambdaSlot}, Lambda functions used to define getter and setter logic require the
+ * owner's `Scriptable` object as one of the parameters. This is particularly useful for
+ * implementing properties that behave like standard JavaScript properties, but are implemented with
+ * native functionality without the need for reflection.
+ */
+public class LambdaAccessorSlot extends Slot {
+    private transient Function<Scriptable, Object> getter;
+    private transient BiConsumer<Scriptable, Object> setter;
+    private LambdaFunction getterFunction;
+    private LambdaFunction setterFunction;
+
+    LambdaAccessorSlot(Object name, int index) {
+        super(name, index, 0);
+    }
+
+    LambdaAccessorSlot(Slot oldSlot) {
+        super(oldSlot);
+    }
+
+    @Override
+    boolean isValueSlot() {
+        return false;
+    }
+
+    @Override
+    boolean isSetterSlot() {
+        return true;
+    }
+
+    @Override
+    ScriptableObject getPropertyDescriptor(Context cx, Scriptable scope) {
+        ScriptableObject desc = (ScriptableObject) cx.newObject(scope);
+
+        int attr = getAttributes();
+        boolean es6 = cx.getLanguageVersion() >= Context.VERSION_ES6;
+        if (es6) {
+            if (getterFunction == null && setterFunction == null) {
+                desc.defineProperty(
```

Comment:
Not 100% sure, but doesn't the pd need a 'value' property as well if it won't have getter?

Reply:
Good question. I pretty much followed how [AccessorSlot](https://github.com/mozilla/rhino/blob/b7ece52d83a0d558b28b5962c63e8eea31f0ecd8/rhino/src/main/java/org/mozilla/javascript/AccessorSlot.java#L45) implements this. 

Having said that, trying below

``` js 

> const o = {};
undefined

>Object.defineProperty(o, "b", {
		enumerable: true,
		configurable: true
});
{b: undefined}


> Object.getOwnPropertyDescriptor(o, "b")
{value: undefined, writable: false, enumerable: true, configurable: true}
```

 in both Chrome and nodejs does return `value: undefined` so perhaps we should add it?